app/lib/shared/markdown.dart

```diff
@@ -250,18 +270,15 @@ Iterable<m.Node> _groupChangelogNodes(List<m.Node> nodes) sync* {
   m.Element lastContentDiv;
   for (final node in nodes) {
     final version = (node is m.Element &&
+            _structuralHeaderTags.contains(node.tag) &&
```

Comment:
I think we should also require internal consistency in the changelog file.. ie. it uses only h1 or h2 or h3 for all versions... We should avoid mixing them...

Reply:
I think that's a good check for scoring the package, but I wouldn't block the display of the structured changelog on it.

Reply:
Imagine a change logs as follows:
```md
# 2.1.0
 * Zebras can now encode bar-codes
## Upgrading from
### 1.0.0
 * Elephants are now scared of mice. 
### 0.9.0
 * Take care to feed lion before releasing mice.

-------

# 2.0.0
 * Zebras have been added.
 * Elephants are now scared of mice. 

-------

# 1.0.0
 * Take care to feed lion before releasing mice.
```

In this changelog we have headings made correctly, it just so happens that I use sub-headlines within paragraphs to indicate how to upgrade from previous versions... That's very nice of me to provide such elaborate instructions, but also breaks our parser.

side note: I had lots of fun writing this changelog... I almost want to write the package for it :)

Reply:
The example was way too good, I needed to put that into a test :). I've updated the code to match the first tag that matches a version pattern, subsequent matches are only used as a separate version if the tag level matches. PTAL.